rules/apps.libsonnet

```
@@ -269,6 +294,51 @@
               workload_type: 'job',
             },
           },
+          // workload aggregation for barepods
+          {
+            record: 'namespace_workload_pod:kube_pod_owner:relabel',
+            expr: |||
+              max by(%(clusterLabel)s, namespace, workload, pod) (
+                label_replace(
+                  kube_pod_owner{%(kubeStateMetricsSelector)s, owner_kind=""}
+                ), "workload", "$1", "barepod", "(.+)"
+              )
+            ||| % $._config,
+            labels: {
+              workload_type: 'barepod',
+            },
+          },
+          // workload aggregation for staticpods
+          {
+            record: 'namespace_workload_pod:kube_pod_owner:relabel',
+            expr: |||
+              max by(%(clusterLabel)s, namespace, workload, pod) (
+                label_replace(
+                  kube_pod_owner{%(kubeStateMetricsSelector)s, owner_kind="Node"}
+                ), "workload", "$1", "pod", "(.+)"
+              )
+            ||| % $._config,
+            labels: {
+              workload_type: 'staticpod',
+            },
+          },
+          // workload aggregation for non-standard workloads
+          {
+            record: 'namespace_workload_pod:kube_pod_owner:relabel',
+            expr: |||
+              max by(%(clusterLabel)s, namespace, workload, pod, workload_type) (
+                label_join(
+                  kube_pod_owner{%(kubeStateMetricsSelector)s, owner_kind!~"ReplicaSet|Deployment|DaemonSet|StatefulSet|Job||Node"},
+                  "workload_type", "", "owner_kind"
+                )
+                or
+                label_join(
+                  kube_replicaset_owner{%(kubeStateMetricsSelector)s, owner_kind!~"Deployment"},
```

Couple of points with the right side of this `or`:

1. Need to ignore empty owner_kind (suggested change below)
2. `kube_replicaset_owner` shouldn't be used on its own in these rules which are pod-level in nature, it should only be used when joining against `kube_pod_owner` (as the recording rule name suggests).

```suggestion
                  kube_replicaset_owner{%(kubeStateMetricsSelector)s, owner_kind!="", owner_kind!="Deployment"},
```